src/transaction.js

```diff
@@ -258,11 +276,21 @@ export class Transaction {
       );
     }
 
+    let taggedTransaction;
+
+    if (this.isFeeBump()) {
+      taggedTransaction = new xdr.TransactionSignaturePayloadTaggedTransaction.envelopeTypeTxFeeBump(
+        tx
+      );
+    } else {
+      taggedTransaction = new xdr.TransactionSignaturePayloadTaggedTransaction.envelopeTypeTx(
+        tx
+      );
+    }
+
     const txSignature = new xdr.TransactionSignaturePayload({
       networkId: xdr.Hash.fromXDR(hash(this.networkPassphrase)),
-      taggedTransaction: new xdr.TransactionSignaturePayloadTaggedTransaction.envelopeTypeTx(
-        tx
-      )
+      taggedTransaction
```

Comment:
This is probably me not knowing JS well, but shouldn't this have a key?

Reply:
oh it doesn't - in JS, if the key has the same name as the variable you are assigning, it will get merged into the object.

Reply:
ok, thanks!!